src/context/product-context.jsx

```diff
@@ -0,0 +1,47 @@
+import axios from "axios";
+import { createContext, useContext, useState, useReducer, useEffect } from "react";
+import { inStock, priceRange, rating, categoryName } from "../utilities/filters/filterModule";
+import { productFilterList } from "../utilities/filters/productFilterList";
+import { applyFilters } from "../utilities/filters/applyFilter";
+import { filterSettings } from "../utilities/filters/filterSetting";
+
+const ProductContext = createContext(filterSettings);
+
+const useProducts = () => useContext(ProductContext);
+
+const ProductProvider = ({ children }) => {
+    const [products, setProducts] = useState([]);
+    const [state, dispatch] = useReducer(applyFilters, filterSettings);
+    const [categories, setCategories] = useState([]);
+
+    useEffect(() => {
+        (async () => {
+          try {
+            const res = await axios.get("/api/categories");
+            console.log(res.data.categories)
```

Comment:
You can remove console log statements

Reply:
Thanks for bringing that to notice will remove that.